Aquality.WinAppDriver/src/Aquality.WinAppDriver/Elements/ElementFactory.cs

```diff
@@ -20,29 +21,40 @@ namespace Aquality.WinAppDriver.Elements
     /// </summary>
     public class ElementFactory : CoreFactory, IElementFactory
     {
-        public ElementFactory(ConditionalWait conditionalWait, IElementFinder elementFinder, ILocalizationManager localizationManager) : base(conditionalWait, elementFinder, localizationManager)
+        private readonly Func<ISearchContext> searchContextSupplier;
+        private readonly bool isRootSession;
+
+        public ElementFactory(
+            ConditionalWait conditionalWait, 
+            IElementFinder elementFinder, 
+            ILocalizationManager localizationManager, 
+            Func<ISearchContext> searchContextSupplier = null, 
+            bool isRootSession = false) 
+            : base(conditionalWait, elementFinder, localizationManager)
         {
+            this.searchContextSupplier = searchContextSupplier;
+            this.isRootSession = isRootSession;
         }
 
-        public IButton GetButton(By locator, string name, ElementState elementState = ElementState.Displayed)
+        public virtual IButton GetButton(By locator, string name, ElementState elementState = ElementState.Displayed)
```

Comment:
Are you sure we need this methods as virtual ?

Reply:
yes, for the case of custom buttons on the project where this library is used